client/components/section-nav/index.jsx

```diff
@@ -116,6 +116,8 @@ class SectionNav extends Component {
 			if ( child.type === Search ) {
 				if ( child.props.pinned ) {
 					this.hasPinnedSearch = true;
+
+					extraProps.onSearchOpen = this.generateOnSearchOpen( child.props.onSearchOpen );
```

Comment:
I'd discourage any mutation whatsoever, as it could sometimes lead to unexpected and hard to trace bugs. I realize there are similar pre-existing mutations here, but also perhaps this is a good opportunity to improve them?

Reply:
I will be leaving the decision to remove the mutations for the calypso components team themselves.